### ISSUE TITLE
DTree: move assignment operator

### ISSUE THREAD

**Issue by jvandewync** [mlpack/mlpack#875]
I got the following error while building mlpack dtree_impl.hpp (line 300):
 'mlpack::det::DTree<arma::Mat<double>,int>::operator=': must return a value	mlpack_det

**Comment by rcurtin** [comment]
Hi there,

Can you provide more information about your setup?  i.e. OS, version of mlpack, build log, etc.?

Thanks!

**Comment by jvandewync** [comment]
I've just followed the guide provided by Keon Kim (https://keon.io/mlpack/mlpack-on-windows/). I build mlpack with VS2015 on Windows 10. A part of log: 

```
5>c:\users\jvandewync\documents\projects\mlpack 2.1.1\src\mlpack\methods\det\dtree_impl.hpp(773): warning C4244: 'initializing': conversion from 'double' to 'const std::size_t', possible loss of data
5>c:\users\jvandewync\documents\projects\mlpack 2.1.1\src\mlpack\methods\det\dtree_impl.hpp(775): warning C4244: 'initializing': conversion from 'double' to 'std::size_t', possible loss of data
5>c:\users\jvandewync\documents\projects\mlpack 2.1.1\src\mlpack\methods\det\dtree_impl.hpp(775): warning C4244: 'initializing': conversion from 'double' to 'const std::size_t', possible loss of data
5>c:\users\jvandewync\documents\projects\mlpack 2.1.1\src\mlpack\methods\det\dtree_impl.hpp(776): warning C4244: 'initializing': conversion from 'double' to 'std::size_t', possible loss of data
5>c:\users\jvandewync\documents\projects\mlpack 2.1.1\src\mlpack\methods\det\dtree_impl.hpp(776): warning C4244: 'initializing': conversion from 'double' to 'const std::size_t', possible loss of data
5>c:\users\jvandewync\documents\projects\mlpack 2.1.1\src\mlpack\methods\det\dtree_impl.hpp(300): error C4716: 'mlpack::det::DTree<arma::Mat<double>,int>::operator=': must return a value
```

In order to successfully build mlpack, I have inserted a `return *this;` at the end of the function, is it right?

**Comment by rcurtin** [comment]
Yes, it looks like that was a problem.  I committed that fix in f1701b194.  Let me know if that doesn't fix the issue and we can reopen the bug.  Thanks for the report!
